ui/raidboss/popup-text.ts

```diff
@@ -817,58 +674,235 @@ export class PopupText {
+    if (!haveZoneRegex && !haveZoneId || haveZoneRegex && haveZoneId) {
+      console.error(`Trigger set must include exactly one of zoneRegex or zoneId property`);
+      return false;
+    }
+
+    if (haveZoneId && set.zoneId === undefined) {
+      console.error(
+        `Trigger set has zoneId, but with nothing specified in ${set.filename}.  Did you misspell the ZoneId.ZoneName?`,
+      );
+      return false;
+    }
+
+    if (set.zoneId !== undefined) {
+      if (
+        set.zoneId !== ZoneId.MatchAll && set.zoneId !== this.zoneId &&
+        !(Array.isArray(set.zoneId) && set.zoneId.includes(this.zoneId))
+      )
+        return false;
+    } else if (set.zoneRegex) {
+      let zoneRegex = set.zoneRegex;
+      if (typeof zoneRegex !== 'object') {
+        console.error(
+          'zoneRegex must be translatable object or regexp: ' + JSON.stringify(set.zoneRegex),
+        );
+        return false;
+      }
+
+      if (!(zoneRegex instanceof RegExp)) {
+        const parserLangRegex = zoneRegex[this.parserLang];
+
+        if (parserLangRegex) {
+          zoneRegex = parserLangRegex;
+        } else if (zoneRegex['en']) {
+          zoneRegex = zoneRegex['en'];
+        } else {
+          console.error('unknown zoneRegex parser language: ' + JSON.stringify(set.zoneRegex));
+          return false;
+        }
+
+        if (!(zoneRegex instanceof RegExp)) {
+          console.error('zoneRegex must be regexp: ' + JSON.stringify(set.zoneRegex));
+          return false;
+        }
+      }
+
+      if (this.zoneName.search(Regexes.parse(zoneRegex)) < 0)
+        return false;
+    }
+
+    return true;
+  }
+
+  // process a zone-enabled trigger set
+  // return undefined for bad trigger Set
+  // process trigger based on current job, options and zone.
+  LocalizeTriggerSet(set: LoadedTriggerSet): undefined | LocalizedTrigger[] {
+    // construct something like regexDe or regexFr.
+    const langSuffix = this.parserLang.charAt(0).toUpperCase() + this.parserLang.slice(1);
+
+    const processedTriggers = [];
+
+    if (this.options.Debug) {
+      console.log('Loading ' + set.filename);
+    }
+
+    if (set.initData) {
+      this.dataInitializers.push({
+        file: set.filename,
+        func: set.initData,
+      });
+    }
+
+    // Adjust triggers for the parser language.
+    if (set.triggers && this.options.AlertsEnabled) {
+      for (const [index, trigger] of set.triggers.entries()) {
+        const localized = this.LocalizeTrigger(trigger, set, index, langSuffix);
+        if (localized) {
+          processedTriggers.push(localized);
+        }
+      }
+    }
+
+    return processedTriggers;
+  }
+
+  LocalizeTrigger(
+    trigger: LooseTrigger,
+    set: LoadedTriggerSet,
+    triggerIndex: number,
+    langSuffix: string,
+  ): LocalizedTrigger | undefined {
+    if ('disabled' in trigger && trigger.disabled) {
+      return { id: trigger.id, filename: set.filename, disabled: true };
+    }
```

Comment:
It seems more complicated to add a disabled type to `LocalizedTrigger`.  What about returning `undefined` here instead, or some other way to skip disabled triggers before this is called?

Reply:
disabled trigger may override other trigger so they need to be keeped.

and you are right, need a better aproach here.

Reply:
Oh, nevermind, these need to be included so an override trigger with disabled works.  Got it.